### DISCUSSION TITLE
compile error

### DISCUSSION THREAD

**Discussion by lazy-wangxl** [prometheus/pushgateway#523] [Q&A]
I want to compile from the sources(stable), and i did like this:

`[root@localhost pushgateway]# go mod init

[root@localhost pushgateway]# go mod tidy

[root@localhost pushgateway]# make build`

but failed, error:
_go: module github.com/golang/protobuf is deprecated: Use the "google.golang.org/protobuf" module instead.
>> building binaries
 >   pushgateway
go: inconsistent vendoring in /home/wangxl/WorkSpace/TMT/src/pushgateway:
        github.com/Sirupsen/logrus@v0.8.8-0.20151204141443-446d1c146faa: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/beorn7/perks@v1.0.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/elazarl/go-bindata-assetfs@v0.0.0-20151224045452-57eb5e1fc594: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/golang/protobuf@v1.5.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/julienschmidt/httprouter@v1.3.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/matttproud/golang_protobuf_extensions@v1.0.4: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/prometheus/client_golang@v1.14.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/prometheus/client_model@v0.3.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/prometheus/common@v0.39.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/prometheus/procfs@v0.8.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/sys@v0.3.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/Masterminds/semver@v1.5.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/alecthomas/template@v0.0.0-20190718012654-fb15b899a751: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/alecthomas/units@v0.0.0-20211218093645-b94a6e3cc137: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/cespare/xxhash/v2@v2.2.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/google/go-github/v25@v25.1.3: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/google/go-querystring@v1.1.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/pkg/errors@v0.9.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/prometheus/promu@v0.14.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        go.uber.org/atomic@v1.10.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/net@v0.4.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/oauth2@v0.3.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        google.golang.org/appengine@v1.6.7: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        google.golang.org/protobuf@v1.28.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        gopkg.in/alecthomas/kingpin.v2@v2.2.6: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        gopkg.in/yaml.v2@v2.4.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt

        To ignore the vendor directory, use -mod=readonly or -mod=mod.
        To sync the vendor directory, run:
                go mod vendor
!! command failed: build -o /home/wangxl/WorkSpace/TMT/src/pushgateway/pushgateway -ldflags -X github.com/prometheus/pushgateway/vendor/github.com/prometheus/common/version.Version=0.3.1 -X github.com/prometheus/pushgateway/vendor/github.com/prometheus/common/version.Revision=602f856b0e840cbabc7e4893ea75cf3e9298af3e -X github.com/prometheus/pushgateway/vendor/github.com/prometheus/common/version.Branch=stable -X github.com/prometheus/pushgateway/vendor/github.com/prometheus/common/version.BuildUser=root@localhost.localdomain -X github.com/prometheus/pushgateway/vendor/github.com/prometheus/common/version.BuildDate=20221220-02:27:24  -extldflags '-static' -a -tags netgo github.com/prometheus/pushgateway: exit status 1
make: *** [build] Error 1
_

**Comment by beorn7** [comment]
You seem to have a vendor directory somewhere that interferes.

Very generally, you should not require to run `go mod init` or `go mod tidy`. Simply running `make` should suffice.

I would also recommend to run a recent version of Go.

**Reply by lazy-wangxl** [reply]
go version go1.19.4 linux/amd64
